Tidy up design documentation for signatures

rptb1 opened the pull request:
Move design documentation of signatures to [design.mps.sig](https://github.com/Ravenbrook/mps/blob/branch/2023-03-09/sig-design-tidy/design/sig.txt)

Ensure consistent tagging of signature fields from code.

Fixes #184 .

rptb1 commented:
Executing [proc.review.express](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/review.rst#express)

1. Start time 12:27
2. Entry passed.
3. No code has actually changed (only comments) so we can just work [rule.generic](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/rule.generic.rst) and [rule.style](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/rule.style.rst).
4. Checking time 10 mins.  Logging at 12:45.

rptb1 reviewed (commented):
Executing [proc.review.check](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/review.rst#54-review-checking)

1. m. The link to guide.hex.trans doesn't work on GitHub. rule.generic.achieve
2. m. Say "immediately before" not "just before". rule.generic.clear
3. m. Invalidating is done after a check, so that's incorrect, and should say "immediately after". rule.generic.self.
4. C. Could use RFC-2119 "MUST NOT" etc. for importance.
5. Checking too 10 minutes.
6. Found 3 minor 1 comment.

thejayps commented on design/sig.txt:
```diff
@@ -24,56 +26,77 @@ ways the MPS detects corruption or the passing of illegal data is using
 
-There must also be a definition for the valid value for that signature::
+_`.value`: There must also be a definition for the valid value for
+that signature::
```
m: we do not specify whether sigs should be unique to classes. The use case of run time checking implies they might be, however 2 similarly named classes could end up with identical sigs with different meanings. We should decide if we need a rule about this, either to impose uniqueness or we might equally decide we do not want to impose such a limitation on a developer.

thejayps reviewed (changes requested):
1m

rptb1 commented:
Continuing proc.review.express at proc.review.express.log:

1. Nm. Why link to check.h (source code) rather than design.mps.check (design)?  Historical, I think.
2. Brainstorm: Is there some sort of GitHub document presentation link checker?  i.e. checks links in documents *as presented* by GitHub's "preview" of reStructuredText etc.

rptb1 commented on design/sig.txt:
```diff
@@ -24,56 +26,77 @@ ways the MPS detects corruption or the passing of illegal data is using
 
     typedef struct mps_message_s {
-      Sig sig;                      /* <design/sig/> */
+      Sig sig;                      /* design.mps.sig */
```
Fixed: Updated example in https://github.com/Ravenbrook/mps/pull/193/commits/1b2aab567fcb98f8a529eb113d40893f47df1683

thejayps reviewed (approved):
Approved after changes made after proc.review.express

rptb1 commented:
>     1. m. The link to guide.hex.trans doesn't work on GitHub. rule.generic.achieve

Fixed: Link updated in https://github.com/Ravenbrook/mps/pull/193/commits/1b2aab567fcb98f8a529eb113d40893f47df1683

>     2. m. Say "immediately before" not "just before". rule.generic.clear

Rejected: @thejayps and I didn't think this made things clearer.
 
>     3. m. Invalidating is done after a check, so that's incorrect, and should say "immediately after". rule.generic.self.

Fixed: Clarified design.mps.sig.finish in https://github.com/Ravenbrook/mps/pull/193/commits/1b2aab567fcb98f8a529eb113d40893f47df1683

>     4. C. Could use RFC-2119 "MUST NOT" etc. for importance.

Added a TODO comment in https://github.com/Ravenbrook/mps/pull/193/commits/1b2aab567fcb98f8a529eb113d40893f47df1683

rptb1 commented:
>     1. Nm. Why link to check.h (source code) rather than design.mps.check (design)?  Historical, I think.

Fixed: Linked to deisgn.mps.check.macro.sig in https://github.com/Ravenbrook/mps/pull/193/commits/2697bc1891527902c5f4cb17d6928fc1f57b6e14

>     2. Brainstorm: Is there some sort of GitHub document presentation link checker?  i.e. checks links in documents _as presented_ by GitHub's "preview" of reStructuredText etc.

Forget: Not very important and it'll come up again.

rptb1 commented:
Executing [proc.review.exit](https://github.com/Ravenbrook/mps/blob/586336a7021c2276a3f798ad10e6e4ab5f438226/procedure/review.rst#exit)

1. Defects remaining: One or two minor?
2. Revised change passed.
3. Express review took about 40 minutes (deducting 10 mins for point of order).

rptb1 commented:
Executing [proc.merge.pull-request](https://github.com/Ravenbrook/mps/blob/8635e900f1ca1cbd59b958c10c6b609d9b404457/procedure/pull-request-merge.rst)

1. Start time 20:10.
2. There is no automated test for the problem of signature fields not referencing the design.  It's possible (`grep -E -w -e 'Sig[[:space:]]+sig[[:space:]]*;' code/*.[ch] | grep -v -e 'design.mps.sig.field'`) but I don't think it's worthwhile setting up.
3. Merge took 10 mins.
